caffe2/python/optimizer.py

```diff
@@ -177,6 +176,70 @@ def scale_learning_rate(self, scale):
         self.base_learning_rate *= scale
         return
 
+class MultiPrecisionSgdOptimizer(SgdOptimizer):
+    def __init__(self, base_learning_rate, momentum, policy, **kwargs):
+        super(SgdOptimizer, self).__init__() # self, base_learning_rate, **kwargs)
+        self.base_learning_rate = base_learning_rate
+        self.momentum = momentum
+        self.policy = policy
+        self.init_kwargs = kwargs
+
+    def _run(self, net, param_init_net, param_info):
+        param = param_info.blob
+        param_fp32 = param_info.blob_copy['float'] if param_info.blob_copy is not None else None
+
+        # If we have a straight fp32 parameter, run the base class
+        if param_fp32 == None:
+            return SgdOptimizer._run(self, net, param_init_net, param_info)
+
+        grad = param_info.grad
+        if self.base_learning_rate <= 0:
+            return
+
+        lr, _ = self.build_lr(
+            net, param_init_net,
+            base_learning_rate=-self.base_learning_rate,
+            learning_rate_blob=str(param) + "_lr",
```

Comment:
why do you need str() on param here?  when you sum up an object and a string it would call __str__ on it automatically